flux-driver/src/collector.rs

```diff
@@ -358,11 +373,42 @@ impl<'tcx, 'a> SpecCollector<'tcx, 'a> {
             ("ignore", AttrArgs::Empty) => FluxAttrKind::Ignore,
             ("opaque", AttrArgs::Empty) => FluxAttrKind::Opaque,
             ("trusted", AttrArgs::Empty) => FluxAttrKind::Trusted,
+            ("extern_spec", AttrArgs::Empty) => FluxAttrKind::ExternSpec,
             _ => return Err(self.emit_err(errors::InvalidAttr { span: attr_item.span() })),
         };
         Ok(FluxAttr { kind, span: attr_item.span() })
     }
 
+    // In Prusti they suggested looking into doing this instead of using a Visitor...
+    // it seems more brittle but I guess conversely their version is a little permissive.
+    fn extract_extern_def_id_from_extern_spec_fn(
+        &mut self,
+        def_id: LocalDefId,
+    ) -> Result<DefId, ErrorGuaranteed> {
+        use rustc_hir::{def, ExprKind, Node};
+        if let Node::Item(i) = self.tcx.hir().find_by_def_id(def_id).unwrap() {
+            if let ItemKind::Fn(_rust_fn_sig, _generics, body_id) = &i.kind {
```

Comment:
I typically avoid defining names if we are not going to use them to avoid the extra noise.

```suggestion
            if let ItemKind::Fn(_, _, body_id) = &i.kind {
```